### ISSUE TITLE
Can't open replays recorded on a server 1.20.1

### ISSUE THREAD

**Issue by Igneo220** [ferriarnus/ReForgedPlay#13]
there are a lot of mods installed, but to summarize, it works perfectly in singleplayer, but recordings in the server cannot open

![image](https://github.com/user-attachments/assets/be636c3a-1382-447a-8f8a-d7a808ffd2f5)

It gives me whatever this error is, and a lot of array index out of bounds errors too

[latest.log](https://github.com/user-attachments/files/16631354/latest.log)

**Comment by ferriarnus** [comment]
I see nothing in the log that is obivous. But why does it have fabric mods without connector installed?

**Comment by Igneo220** [comment]
I haven't installed any fabric mods. What do you mean?

**Comment by ferriarnus** [comment]
The log contains fabric api's, though these might be from FFRAPI. I'll test it on a server later.

**Comment by EnderKingDubs** [comment]
Hey, first off I want to thank you for undertaking the task of porting Replay to Forge. It's been painful wanting to use forge mods and only having Fabric options for 3rd person camera work, so I can't thank you enough.

But I'm experiencing this same issue, literally made this account just to share. Here's my log, if you need any other info that might make things easier let me know and I'll try to provide.

[latest.log](https://github.com/user-attachments/files/16812193/latest.log)

**Comment by whatamaniac** [comment]
I'm facing the same issue while opening any replays recorded on multiplayer servers on Forge 1.20.1, while replays recorded on singleplayer still work perfectly fine. I'm using it on Integrated Minecraft Modpack.
![image](https://github.com/user-attachments/assets/f669caff-2f4b-4279-a735-8de79aa682d3)
Is there any temporary fix for this? or any suspicion on any mod which can cause this? Here's my Latest log.
[latest.log](https://github.com/user-attachments/files/17868363/latest.log)

**Comment by whatamaniac** [comment]
Also i had redirector mod in my modpack but i removed it to see if that's the mod which is actually giving this issue but unfortunately it's not.

**Comment by Igneo220** [comment]
rn theres no known fix and believe me i spent hours looking for one.

For some of my server members it works fine, and then for the same people later it then completely broke and they have the same issue, and other members like me had it break instantly.

Occasionally it'll go past this screen and put me in limbo world as if it was loading something, but checking the logs reveals its recursively running the same code.

What's interesting to note is that if you use sinytra connector with the main branch of replay you get the same issue.

So like all good coding issues, its unpredictable, unsolved, only happens to specific people, and the log spits out unusable garbage. We love modding minecraft

**Comment by whatamaniac** [comment]
> rn theres no known fix and believe me i spent hours looking for one.
> 
> For some of my server members it works fine, and then for the same people later it then completely broke and they have the same issue, and other members like me had it break instantly.
> 
> Occasionally it'll go past this screen and put me in limbo world as if it was loading something, but checking the logs reveals its recursively running the same code.
> 
> What's interesting to note is that if you use sinytra connector with the main branch of replay you get the same issue.
> 
> So like all good coding issues, its unpredictable, unsolved, only happens to specific people, and the log spits out unusable garbage. We love modding minecraft

Can you share your modlist with me? (if you still have it) so that i could try to debug it. I do think it is some mod that's causing this issue.. Not sure if it's sinytra connector. I'll test with that also.

**Comment by ferriarnus** [comment]
Hmm, I'm unable to recreate this using a forge server and clients. I there a minimal reproduction pack for this?

**Comment by jessentyx** [comment]
also running into  the  same problem, have  anyone  found the solution  to  this?

**Comment by Brickmaster1** [comment]
I am having the same problem in a pack I am playing. 

[2025-08-26-1.log.gz](https://github.com/user-attachments/files/21998874/2025-08-26-1.log.gz)

**Comment by ferriarnus** [comment]
Quick question, did you open a modded screen on the server? If so give https://github.com/ferriarnus/ReForgedPlay/actions/runs/17344286627 a try

**Comment by Brickmaster1** [comment]
> Quick question, did you open a modded screen on the server? If so give https://github.com/ferriarnus/ReForgedPlay/actions/runs/17344286627 a try

I have opened quite a few.. I thought replay didn't record screens? I am testing now to see if this works, but I  have definitely opened like Mekanism machines, Create machines and items like filters, schematic(cannon)s..

**Comment by Brickmaster1** [comment]
Nope, I sitll get the same problem with the weird text, and the log spam with `java.lang.IndexOutOfBoundsException: Index 340 out of bounds for length 111`

**Comment by ferriarnus** [comment]
Hmm, guess that's not the issue then. I'll keep looking.

**Comment by T0biasCZe** [comment]
Having the same issue (In Create Ultimate Selection modpack), throws the "Failed to connect to server: <?>[BEL]+f" error in gui, and in log it spams the java.lang.IndexOutOfBoundsException: Index X out of bounds for length 111

attaching the log and short sample replay if it may help...

https://github.com/user-attachments/files/23145799/PjZkWR6W.txt        
https://tobikcze.eu/files/2025_10_26_02_56_55.mcpr

**Comment by BeckPhillips** [comment]
I'm having the same issue, with the same error and recursive code loop in my log >.>
I'm playing on a modded Forge 1.20.1 server with a bunch of friends, with a few fabric mods, running sinytra connector.

<img width="964" height="526" alt="Image" src="https://github.com/user-attachments/assets/54776c10-72c7-44a8-a2bf-26830de41a8d" />

The "full" log is way too long to post, so I chopped out about 98% of it, which were the exact same set of lines - it seems to be referencing "Index X out of bounds for length 111."
https://pastebin.com/sifgJnPS

As a brief note - every other Forge port of ReplayMod has this exact problem, which leads me to wonder if it's an issue with the mod itself, rather than Forge... but I have no idea what I'm talking about, just trying to offer solutions that probably don't help.

**Comment by Brickmaster1** [comment]
Yes, even using Sinytra Connector with fabric replay exhibits the same problem (on the last compatible version you can do this with).

**Comment by pipavlo82** [comment]
Built a small demo around match dispute resolution:

- valid chain -> PASS
- tampered event -> FAIL
- replay injected -> FAIL

Core idea:
we resolve match disputes with verifiable execution receipts.

Curious if something like this would be useful in your flow?
